src/krb5_plugin/passkey/passkey_utils.c

```diff
@@ -602,33 +602,46 @@ sss_passkey_concat_credentials(char **creds,
 {
     krb5_error_code ret;
     char *result_creds = NULL;
-    char *tmp_creds = NULL;
-    size_t creds_len = 0;
+    size_t total_sz = 0;
+    size_t creds_buf_len = 0;
+    size_t len = 0;
+    int rc = 0;
 
-    result_creds = strdup(creds[0]);
+    for (int i = 0; creds[i] != NULL; i++) {
+        total_sz += strlen(creds[i]);
+    }
+    total_sz++;
+
+    result_creds = malloc(total_sz);
     if (result_creds == NULL) {
-        ret = ENOENT;
+        ret = ENOMEM;
         goto done;
     }
 
-    creds_len += strlen(creds[0] + 1);
+    creds_buf_len = strlen(creds[0]);
 
-    for (int i = 1; creds[i] != NULL; i++) {
-        strcat(result_creds, ",");
-        creds_len += strlen(creds[i]) + 1;
+    rc = snprintf(result_creds, total_sz, "%s", creds[0]);
+    if (rc < 0 || rc > creds_buf_len) {
+        ret = ENOMEM;
+        free(result_creds);
+        goto done;
+    }
+
+    len += creds_buf_len;
 
-        tmp_creds = realloc(result_creds, creds_len);
-        if (tmp_creds == NULL) {
+    for (int i = 1; creds[i] != NULL; i++) {
+        rc = snprintf(result_creds + len, total_sz - len, ",%s", creds[i]);
```

Comment:
Be careful, you are not counting the comma when allocating memory and the buffer may overflow.

Reply:
Fixed, thanks.